Dockerfile

```diff
@@ -52,4 +52,9 @@ FROM base as test
 
 # Set environment variables
 ENV FLASK_ENV=test
-ENTRYPOINT [ "/venv/bin/poetry", "run","pytest"]
\ No newline at end of file
+
+# Run dependency vulnerability check using safety
+RUN /venv/bin/poetry run safety check
```

Comment:
Ooh I don't quite like the look of this! While this does currently work, having the safety check run at build time feels a little odd. In particular, this means that when I'm building my Docker image to run unit tests, I won't be able to run the tests if there's a vulnerability (since I can't complete the build); we're tying those two things together very tightly whereas I'd rather they were more loosely coupled

I think much nicer to introduce this as an alternative entrypoint so that it can fire when we run the docker container. E.g. something like:
```Dockerfile
# Testing stage
FROM base as test

# Set environment variables
ENV FLASK_ENV=test

# Run tests using pytest
ENTRYPOINT ["/venv/bin/poetry", "run", "pytest"]

FROM test as securityscan
ENTRYPOINT ["/venv/bin/poetry", "run", "safety", "check"]
```
And then we can run the below to perform our scan:
```bash
docker build --target securityscan --tag todo-app:security .
docker run todo-app:security
```